x-pack/plugin/repository-encrypted/src/main/java/org/elasticsearch/repositories/encrypted/EncryptedRepository.java

```diff
@@ -6,11 +6,682 @@
+import java.security.GeneralSecurityException;
+import java.security.SecureRandom;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class EncryptedRepository extends BlobStoreRepository {
+    static final Logger logger = LogManager.getLogger(EncryptedRepository.class);
+    // the following constants are fixed by definition
     static final int GCM_TAG_LENGTH_IN_BYTES = 16;
     static final int GCM_IV_LENGTH_IN_BYTES = 12;
-    static final int AES_BLOCK_SIZE_IN_BYTES = 128;
+    static final int AES_BLOCK_LENGTH_IN_BYTES = 128;
+    // the following constants require careful thought before changing because they will break backwards compatibility
     static final String DATA_ENCRYPTION_SCHEME = "AES/GCM/NoPadding";
     static final long PACKET_START_COUNTER = Long.MIN_VALUE;
-    static final int MAX_PACKET_LENGTH_IN_BYTES = 1 << 30;
+    static final int MAX_PACKET_LENGTH_IN_BYTES = 8 << 20; // 8MB
+    // this should be smaller than {@code #MAX_PACKET_LENGTH_IN_BYTES} and it's what {@code EncryptionPacketsInputStream} uses
+    // during encryption and what {@code DecryptionPacketsInputStream} expects during decryption (it is not configurable)
+    static final int PACKET_LENGTH_IN_BYTES = 64 * (1 << 10); // 64KB
+    // the path of the blob container holding all the DEKs
+    // this is relative to the root base path holding the encrypted blobs (i.e. the repository root base path)
+    static final String DEK_ROOT_CONTAINER = ".encryption-metadata"; // package private for tests
+    static final int DEK_ID_LENGTH = 22; // {@code org.elasticsearch.common.UUIDS} length
+
+    // the following constants can be changed freely
+    private static final String RAND_ALGO = "SHA1PRNG";
+    // each snapshot metadata contains an unforgeable identifier of the repository password of the master node that started the snapshot
+    // this hash is then verified on each data node before the actual shard files snapshot, as well as on the
+    // master node that finalizes the snapshot (could be a different master node, if a master failover occurred during the snapshot)
+    static final String PASSWORD_ID_USER_METADATA_KEY = EncryptedRepository.class.getName() + ".repositoryPasswordId";
+    static final String PASSWORD_ID_SALT_USER_METADATA_KEY = EncryptedRepository.class.getName() + ".repositoryPasswordIdSalt";
+    private static final int DEK_CACHE_WEIGHT = 2048;
+
+    // this is the repository instance to which all blob reads and writes are forwarded to (it stores both the encrypted blobs, as well
+    // as the associated encrypted DEKs)
+    private final BlobStoreRepository delegatedRepository;
+    // every data blob is encrypted with its randomly generated AES key (DEK)
+    private final Supplier<Tuple<BytesReference, SecretKey>> DEKGenerator;
```

Comment:
Yeah, I think this is the code format convention we generally use.

There are many other variable names that start with caps because the first word is an acronym , and I've renamed them all.